lib/pos_mempool_transaction.go

```diff
@@ -26,6 +26,9 @@ type MempoolTx struct {
 	// The time when the txn was added to the pool
 	Added time.Time
 
+	// Whether this transaction has been validated by the mempool
+	Validated bool
```

Comment:
Since we have getters and setters for this, shouldn't we make it private by making it lowercase?